qiskit/providers/ibmq/managed/managedjobset.py

```diff
@@ -418,6 +419,42 @@ def name(self) -> str:
         """
         return self._name
 
+    def update_name(self, name: str) -> str:
```

Comment:
Also, `ManagedJobSet.retrieve_jobs()` verifies each job name is proper (i.e. all have the same prefix). Since individual job can now have its name updated at random, it may be easier to just not do that verification.